### PR TITLE
Add TPU bfloat16 Support for Semantic Segmentation Script

### PR DESCRIPTION
**Pull request by DavidLandup0**
Continuing #1183, as a series of PRs as discussed in comments to update the sem seg training script into a general one.

Roadmap:
[This PR] Support TPU with mixed precision (currently only works on GPUs)
* [ ] Epochs flag
* [ ] Support backbones via flags [requires running a test]
* [ ] Change preprocessing functions to use KCV layers and simplify the calls [requires running a test]
* [ ] Loss function flag
* [ ] Optimizer flag
* [ ] LR Schedule flag
* [ ] EMA and weight decay (2.11)

@tanzhenyu

### CODE REVIEW

**Comment by DavidLandup0** [comment]
Do we have a public Pascal VOC on a GCS bucket for the TPU to work with?
Afaik, the current script downloads them locally, so a TPU wouldn't help. I might help debug the slowdown issue

**Comment by tanzhenyu** [comment]
> Do we have a public Pascal VOC on a GCS bucket for the TPU to work with? Afaik, the current script downloads them locally, so a TPU wouldn't help. I might help debug the slowdown issue

We don't have a public bucket, everything is done manually for now.

**Comment by tanzhenyu** [comment]
Is this still WIP? We could close it if not

**Comment by DavidLandup0** [comment]
Still WIP - I just haven't gotten a hold of a GCS to test this yet.
I'll look into what might've prevented efficient TPU utilisation today/tomorrow since there are other elements for updating in the training script

**Comment by ianstenbit** [comment]
Hey @DavidLandup0 -- sorry this got buried.

Are you still planning to work on these segmentation scripts?
If so, feel free to assign this PR to me to review, otherwise let's close it for now.

**Comment by LukeWood** [comment]
Closing due to inactivity!  Feel free to reopen.
